sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala

```diff
@@ -0,0 +1,296 @@
+      (df.parse(p.timestamp).getTime, withNumberInvalid { p.processedRowsPerSecond })),
+      Array.empty[(Long, Double)])
+    val inputRowsData = withNoProgress(query, query.recentProgress.map(p =>
+      (df.parse(p.timestamp).getTime, withNumberInvalid { p.numInputRows })),
+      Array.empty[(Long, Double)])
+    val batchDurations = withNoProgress(query, query.recentProgress.map(p =>
+      (df.parse(p.timestamp).getTime, withNumberInvalid { p.batchDuration })),
+      Array.empty[(Long, Double)])
+    val operationDurationData = withNoProgress(query, query.recentProgress.map { case p =>
+      val durationMs = p.durationMs
+      durationMs.remove("triggerExecution")
+      (df.parse(p.timestamp).getTime, durationMs)},
+      Array.empty[(Long, ju.Map[String, JLong])])
+    val operationLabels = withNoProgress(query, {
+        val durationKeys = query.lastProgress.durationMs.keySet()
+        // remove "triggerExecution" as it count the other operation duration.
+        durationKeys.remove("triggerExecution")
+        durationKeys.asScala.toSeq.sorted
+      }, Seq.empty[String])
+
+    val jsCollector = new JsCollector
+    val graphUIDataForInputRate =
+      new GraphUIData(
+        "input-rate-timeline",
+        "input-rate-histogram",
+        inputRateData,
+        minBatchTime,
+        maxBatchTime,
+        minRecordRate,
+        maxRecordRate,
+        "records/sec")
+    graphUIDataForInputRate.generateDataJs(jsCollector)
+
+    val graphUIDataForProcessRate =
+      new GraphUIData(
+        "process-rate-timeline",
+        "process-rate-histogram",
+        processRateData,
+        minBatchTime,
+        maxBatchTime,
+        minProcessRate,
+        maxProcessRate,
+        "records/sec")
+    graphUIDataForProcessRate.generateDataJs(jsCollector)
+
+    val graphUIDataForInputRows =
+      new GraphUIData(
+        "input-rows-timeline",
+        "input-rows-histogram",
+        inputRowsData,
+        minBatchTime,
+        maxBatchTime,
+        minRows,
+        maxRows,
+        "records")
+    graphUIDataForInputRows.generateDataJs(jsCollector)
+
+    val graphUIDataForBatchDuration =
+      new GraphUIData(
+        "batch-duration-timeline",
+        "batch-duration-histogram",
+        batchDurations,
+        minBatchTime,
+        maxBatchTime,
+        minBatchDuration,
+        maxBatchDuration,
+        "ms")
+    graphUIDataForBatchDuration.generateDataJs(jsCollector)
+
+    val graphUIDataForDuration =
+      new GraphUIData(
+        "duration-area-stack",
+        "",
+        Seq.empty[(Long, Double)],
+        0L,
+        0L,
+        0L,
+        0L,
+        "ms")
+
+    val table =
+    // scalastyle:off
+      <table id="stat-table" class="table table-bordered" style="width: auto">
+        <thead>
+          <tr>
+            <th style="width: 160px;"></th>
+            <th style="width: 492px;">Timelines</th>
+            <th style="width: 350px;">Histograms</th></tr>
```

Comment:
Should we use auto width here?
![image](https://user-images.githubusercontent.com/4833765/68624235-595be900-0511-11ea-915d-17b85ad70757.png)